src/ows/src/main/java/org/geoserver/ows/util/OwsUtils.java

```diff
@@ -420,8 +420,10 @@ else if ( Set.class.isAssignableFrom( type ) ) {
     static void updateCollectionProperty(Object object, Collection newValue, Method getter)
             throws Exception {
         Collection oldValue = (Collection) getter.invoke(object, null);
-        oldValue.clear();
-        oldValue.addAll(newValue);
+        if (oldValue != null) {
+            oldValue.clear();
+            oldValue.addAll(newValue);
```

Comment:
This would avoid setting the newValue? Not sure you are meeting the java bean getter/setter api contract for oldValue?